pkg/db/migration/convert/convert000034/convert.go

```diff
@@ -0,0 +1,160 @@
+package convert000034
+
+import (
+	"bytes"
+	"fmt"
+
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+	"gorm.io/gorm"
+
+	"github.com/instill-ai/pipeline-backend/pkg/datamodel"
+	"github.com/instill-ai/pipeline-backend/pkg/db/migration/convert"
+)
+
+const batchSize = 100
+
+type RenameHTTPComponent struct {
+	convert.Basic
+}
+
+func (c *RenameHTTPComponent) Migrate() error {
+	if err := c.migratePipeline(); err != nil {
+		return err
+	}
+
+	return c.migratePipelineRelease()
+}
+
+func (c *RenameHTTPComponent) migratePipeline() error {
+	pipelines := make([]*datamodel.Pipeline, 0, batchSize)
+	return c.DB.Select("uid", "recipe_yaml").FindInBatches(&pipelines, batchSize, func(tx *gorm.DB, _ int) error {
```

Comment:
nit: instead of searching the entire table, I prefer to [filter out](https://github.com/instill-ai/pipeline-backend/blob/main/pkg/db/migration/convert/convert000031/convert.go#L88) the recipes that don't contain the target component.

Reply:
fixed